source/base/process_grid.cc

```diff
@@ -72,7 +74,7 @@ namespace
     // but this affects the grid shape dramatically, i.e. 10 cores 3x3 becomes
     // 2x5.
     // limit our estimate to be in [2, Np]
-    int n_process_columns = std::min(Np, std::max(2, Pc));
+    int n_process_columns = std::clamp(Pc, 2, Np);
```

Comment:
I think we had this debate in the other PR that dealt with `std::clamp` already: In some cases, `2>Np` and the `std::clamp` function aborts in that case. I don't recall whether we looked into the reason for this, but that part of the patch was reverted because we actually wanted to existing behavior.